examples/oft-solana/tasks/solana/createOFT.ts

```diff
@@ -1,147 +1,202 @@
+            symbol,
+            tokenMetadataIsMutable: isMutable,
+            tokenProgram: tokenProgramStr,
+            uri,
+        }: CreateOFTTaskArgs) => {
+            const isMABA = !!mintStr
+            if (tokenProgramStr !== TOKEN_PROGRAM_ID.toBase58() && !isMABA) {
+                throw new Error('Non-Mint-And-Burn-Adapter does not support custom token programs')
             }
-        )
-        const ixs = [toWeb3JsInstruction(initOftIx.instruction)]
-        const signers = [umiWalletKeyPair, lockBox]
-
-        const txResult = await sendAndConfirmTx(connection, signers, ixs)
-        console.log(`initOFT transaction hash: ${txResult.hash}`)
-
-        await setAuthority(umi, {
-            owned: token.publicKey,
-            owner: umiWalletSigner,
-            newAuthority: fromWeb3JsPublicKey(multiSigKey),
-            authorityType: 0,
-        }).sendAndConfirm(umi, { confirm: { commitment: 'confirmed' } })
-
-        await setAuthority(umi, {
-            owned: token.publicKey,
-            owner: umiWalletSigner,
-            newAuthority: fromWeb3JsPublicKey(multiSigKey),
-            authorityType: 1,
-        }).sendAndConfirm(umi, { confirm: { commitment: 'confirmed' } })
-
-        const outputDir = `./deployments/${endpointIdToNetwork(taskArgs.eid)}`
-        if (!existsSync(outputDir)) {
-            mkdirSync(outputDir, { recursive: true })
-        }
-
-        // Write the JSON file to the specified directory
-        writeFileSync(
-            `${outputDir}/OFT.json`,
-            JSON.stringify(
-                {
-                    programId: taskArgs.programId,
-                    mint: mintPK,
-                    escrow: escrowPK,
-                    oftStore: oftStorePda,
-                },
-                null,
-                4
+            if (isMABA && amount) {
+                throw new Error('Mint-And-Burn-Adapter does not support minting tokens')
+            }
+            const tokenProgramId = publicKey(tokenProgramStr)
+            const { connection, umi, umiWalletKeyPair, umiWalletSigner } = await deriveConnection(eid)
+            const { programId, lockBox, escrowPK, oftStorePda, eddsa } = deriveKeys(programIdStr)
+            const mintAuthorityPublicKey = await createMintAuthorityMultisig(
+                connection,
+                toWeb3JsKeypair(umiWalletKeyPair),
+                toWeb3JsPublicKey(oftStorePda),
+                toWeb3JsPublicKey(tokenProgramId) // Only configurable for MABA
             )
-        )
-        console.log(`Accounts have been saved to ${outputDir}/OFT.json`)
-    })
+            console.log(`created SPL multisig @ ${mintAuthorityPublicKey.toBase58()}`)
+
+            const mint = isMABA
+                ? createNoopSigner(publicKey(mintStr))
+                : createSignerFromKeypair(umi, eddsa.generateKeypair())
+            const isTestnet = eid == EndpointId.SOLANA_V2_TESTNET
+            if (!isMABA) {
```

Comment:
I'd rather see these if/elses being pulled into two separate tasks or task action implementations, the difference in the branch logic is quite large

Reply:
ditto